safe_eth/safe/api/base_api.py

```diff
@@ -16,52 +16,74 @@ class SafeAPIException(Exception):
 
 
 class SafeBaseAPI(ABC):
-    URL_BY_NETWORK: Dict[EthereumNetwork, str] = {}
-
     def __init__(
         self,
         network: EthereumNetwork,
         ethereum_client: Optional[EthereumClient] = None,
         base_url: Optional[str] = None,
+        api_key: Optional[str] = None,
         request_timeout: int = 10,
     ):
         """
         :param network: Network for the transaction service
         :param ethereum_client:
         :param base_url: If a custom transaction service is used
+        :param api_key: Api key of authenticated Safe services
         :raises: EthereumNetworkNotSupported
         """
         self.network = network
         self.ethereum_client = ethereum_client
-        self.base_url = base_url or self.URL_BY_NETWORK.get(network, "")
-        if not self.base_url:
+        base_url = base_url or self.get_url_by_network(network)
```

Comment:
If the purpose of `get_url_by_network` is just fill base_url then I think that the method should be private.